backend/backend/core/chains.py

```diff
@@ -1,17 +1,19 @@
 
+from langchain_core.output_parsers import PydanticOutputParser
 from langchain_core.prompts import ChatPromptTemplate
 from langchain_core.output_parsers import StrOutputParser
-
 from utils.logger import logger
 
 from .models import ModelRegistry
 from .tools import ToolRegistry
 
+from pydantic import BaseModel
+from typing_extensions import List
+
 
 class JokeChain(object):
     def __init__(self, model):
         super().__init__()
-        model = model
         prompt = ChatPromptTemplate.from_template(
```

Comment:
The line `model = model` does not have any effect. The parameter `model` is being shadowed by the local variable of the same name.